### PR TITLE
Fix summary bug: remove target message & refine update strategy

### PR DESCRIPTION
**Pull request by yanjiaxin534**
fix this bug https://github.com/eclipse-symphony/symphony/issues/378 first part: status.targets.xxx's error message won't be reset if current deployment is succeeded.
after this code change, when it change status from failed to succeed, it shows:
image

after this code change, when it change status from failed to succeed, it shows:
image

### CODE REVIEW

File: api/pkg/apis/v1alpha1/model/summary_test.go [hunk 1]
```diff
@@ -34,31 +34,96 @@ func TestUpdateTargetResult(t *testing.T) {
...
+				Message: "Component 1 is bad config",
 			},
 		},
 	})
 	assert.Equal(t, "ERROR", s.TargetResults["target1"].Status)
 	assert.Equal(t, v1alpha2.BadConfig, s.TargetResults["target1"].ComponentResults["component1"].Status)
-	assert.Equal(t, "Component 1 is in bad config", s.TargetResults["target1"].ComponentResults["component1"].Message)
-	assert.Equal(t, 0, s.SuccessCount) //ver 0.48.1: UpdateTargetResult no longer updates success count
+	assert.Equal(t, "Component 1 is bad config", s.TargetResults["target1"].ComponentResults["component1"].Message)
+
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/model/summary_test.go:128]: Why you remove the successCount asserts?

Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/model/summary_test.go:128]: Why you remove the successCount asserts?

**Comment by msftcoderdjw** [comment]
Please take a screenshot how current status looks like after you remove the duplicated target message?

File: api/pkg/apis/v1alpha1/model/summary.go [hunk 2]
```diff
@@ -20,7 +21,6 @@ type ComponentResultSpec struct {
 }
 type TargetResultSpec struct {
 	Status           string                         `json:"status"`
-	Message          string                         `json:"message,omitempty"`
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/model/summary.go:23]: @Haishi2016 , we are going to remove the target.message which is duplicated with component result. Besides the changes in symphony-api, will this affect the OSS portal?

File: api/pkg/apis/v1alpha1/model/summary_test.go (hunk 1 above)
Comment (yanjiaxin534) [reply api/pkg/apis/v1alpha1/model/summary_test.go:128]: because of the comments :ver 0.48.1: UpdateTargetResult no longer updates success count

**Comment by yanjiaxin534** [comment]
> Please take a screenshot how current status looks like after you remove the duplicated target message?

Thanks for comments. Add now

File: api/pkg/apis/v1alpha1/model/summary.go [hunk 3]
```diff
@@ -52,22 +52,21 @@ func (s *SummarySpec) UpdateTargetResult(target string, spec TargetResultSpec) {
...
+				}
 			}
-			message += spec.Message
 		}
 		v.Status = status
-		v.Message = message
-		maps.Copy(v.ComponentResults, spec.ComponentResults)
 		s.TargetResults[target] = v
 	}
+	fmt.Printf("spec status %v", spec)
```
Comment (RemindD) [inline api/pkg/apis/v1alpha1/model/summary.go:69]: Remove debug log?

File: api/pkg/apis/v1alpha1/model/summary.go [hunk 4]
```diff
@@ -52,22 +52,21 @@ func (s *SummarySpec) UpdateTargetResult(target string, spec TargetResultSpec) {
...
 		if spec.Status != "OK" {
 			status = spec.Status
-		}
-		message := v.Message
-		if spec.Message != "" {
-			if message != "" {
-				message += "; "
+		} else {
+			for _, componentStatus := range v.ComponentResults {
+				if componentStatus.Status != v1alpha2.Accepted {
```
Comment (RemindD) [inline api/pkg/apis/v1alpha1/model/summary.go:61]: What is meaning of accepted status and why do we need an exception for it?

File: api/pkg/apis/v1alpha1/model/summary.go [hunk 5]
```diff
@@ -52,22 +52,21 @@ func (s *SummarySpec) UpdateTargetResult(target string, spec TargetResultSpec) {
...
+					status = v.Status
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/model/summary.go:62]: We shouldn't override the status according to component result. The target status is determined by caller.

**Pull request closed without merging**